client/main.cpp

```diff
@@ -406,6 +411,81 @@ int boinc_main_loop() {
     return finalize();
 }
 
+//
+//
+//
+int connect_sock () {
+
+    int portno = 31416;
+    struct sockaddr_in serveraddr;
+    struct hostent *server;
+    char *hostname = "127.0.0.1";
+    int sockfd;
+
+    server = gethostbyname(hostname);
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    serveraddr.sin_family = AF_INET;
+    serveraddr.sin_port = htons(portno);
+    serveraddr.sin_addr = *((struct in_addr *)server->h_addr);
+    bzero(&(serveraddr.sin_zero), 8);
+
+    int n = connect(sockfd, (struct sockaddr *) &serveraddr, sizeof(struct sockaddr));
+    if (n < 0) {
+       printf("The connect error is : %s", strerror(errno));
+    }
+    return sockfd;
+}
+//
+//
+//
+
+//
+//
+//
+void init_websocket() {
+
+    std::vector<std::thread *> threads(/*std::thread::hardware_concurrency()*/1);
+    std::transform(threads.begin(), threads.end(), threads.begin(), [](std::thread *t) {
+        return new std::thread([]() {
+
+
+            uWS::Hub h;
+
+            h.onMessage([](uWS::WebSocket<uWS::SERVER> *ws, char *message, size_t length, uWS::OpCode opCode) {
+            int sockfd = connect_sock();
+            char buf[50000];
+            memset(buf, 0, sizeof(buf));
+            snprintf(buf, /*sizeof(buf)*/ (int) length + 1, "%s", /*(int) msg.len,*/ message);
```

Comment:
possible buffer overflow. you buf variable is 50000 bytes long but you use input variable length that can be more than 50000